client/my-sites/checkout/cart/cart-free-user-plan-upsell.tsx

```diff
@@ -17,186 +18,144 @@ import {
-		const isLoadingSitePlans = this.props.isSitePlansListFetching;
-		return isLoadingCart || isLoadingPlans || isLoadingSitePlans;
-	}
-
-	isRegistrationOrTransfer = ( item: ResponseCartProduct ) => {
-		return isDomainRegistration( item ) || isDomainTransfer( item );
-	};
-
-	getUpgradeText() {
-		const { cart, planPrice, translate, upsellPlan } = this.props;
-		const firstDomain = cart.products.find( this.isRegistrationOrTransfer );
-		const planName = upsellPlan?.getTitle() ?? '';
+const useUpsellPlanSlug = (): PlanSlug => {
+	return PLAN_PERSONAL;
+};
 
-		if ( planPrice && firstDomain && planPrice > firstDomain.cost ) {
-			const extraToPay = planPrice - firstDomain.cost;
-			return translate(
-				'Pay an {{strong}}extra %(extraToPay)s{{/strong}} for our %(planName)s plan, and get access to all its ' +
-					'features, plus the first year of your domain for free.',
-				{
-					args: {
-						extraToPay: formatCurrency( extraToPay, firstDomain.currency ),
-						planName,
-					},
-					components: {
-						strong: <strong />,
-					},
-				}
-			);
-		} else if ( planPrice && firstDomain && planPrice < firstDomain.cost ) {
-			const savings = firstDomain.cost - planPrice;
-			return translate(
-				'{{strong}}Save %(savings)s{{/strong}} when you purchase a WordPress.com %(planName)s plan ' +
-					'instead — your domain comes free for a year.',
-				{
-					args: {
-						planName,
-						savings: formatCurrency( savings, firstDomain.currency ),
-					},
-					components: {
-						strong: <strong />,
-					},
-				}
-			);
-		}
+const isDomainRegistrationOrTransfer = ( item: ResponseCartProduct ) => {
+	return isDomainRegistration( item ) || isDomainTransfer( item );
+};
 
+const UpgradeText = ( {
+	cart,
+	pricingMeta,
+}: {
+	cart: Pick< ResponseCart, 'products' >;
+	pricingMeta: Plans.PricingMetaForGridPlan;
+} ) => {
+	const translate = useTranslate();
+	const upsellPlanSlug = useUpsellPlanSlug();
+	const upsellPlan = getPlan( upsellPlanSlug );
+	const firstDomain = cart?.products?.find( isDomainRegistrationOrTransfer );
+	const planName = upsellPlan?.getTitle() ?? '';
+	const planPrice = pricingMeta.discountedPrice.full || pricingMeta.originalPrice.full;
+
+	if ( planPrice && firstDomain && planPrice > firstDomain.item_subtotal_integer ) {
+		const extraToPay = planPrice - firstDomain.item_subtotal_integer;
 		return translate(
-			'Purchase our %(planName)s plan at {{strong}}no extra cost{{/strong}}, and get access to all its ' +
+			'Pay an {{strong}}extra %(extraToPay)s{{/strong}} for our %(planName)s plan, and get access to all its ' +
 				'features, plus the first year of your domain for free.',
 			{
-				args: { planName },
+				args: {
+					extraToPay: formatCurrency( extraToPay, firstDomain.currency, { isSmallestUnit: true } ),
+					planName,
+				},
+				components: {
+					strong: <strong />,
+				},
+			}
+		);
+	} else if ( planPrice && firstDomain && planPrice < firstDomain.item_subtotal_integer ) {
```

Comment:
Minor but you don't need `else` here because the previous block has a `return`.

```suggestion
	}
	if ( planPrice && firstDomain && planPrice < firstDomain.item_subtotal_integer ) {
```